android/src/main/java/com/rnfingerprint/FingerprintHandler.java

```diff
@@ -0,0 +1,83 @@
+package com.rnfingerprint;
+
+import android.Manifest;
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.hardware.fingerprint.FingerprintManager;
+import android.os.CancellationSignal;
+import android.support.v4.app.ActivityCompat;
+import android.widget.Toast;
+
+public class FingerprintHandler extends FingerprintManager.AuthenticationCallback {
+
+    private CancellationSignal cancellationSignal;
+    private boolean selfCancelled;
+    private Context appContext;
+
+    private final FingerprintManager mFingerprintManager;
+    private final Callback mCallback;
+
+    public FingerprintHandler(Context context, FingerprintManager fingerprintManager, Callback callback) {
+        appContext = context;
+        mFingerprintManager = fingerprintManager;
+        mCallback = callback;
+    }
+
+    public boolean isFingerprintAuthAvailable() {
+        return (android.os.Build.VERSION.SDK_INT >= 23)
+                && mFingerprintManager.isHardwareDetected()
+                && mFingerprintManager.hasEnrolledFingerprints()
+                && (ActivityCompat.checkSelfPermission(appContext,
+                        Manifest.permission.USE_FINGERPRINT) ==
+                        PackageManager.PERMISSION_GRANTED);
+    }
+
+    public void startAuth(FingerprintManager.CryptoObject cryptoObject) {
+
+        cancellationSignal = new CancellationSignal();
+        selfCancelled = false;
+
+        if (ActivityCompat.checkSelfPermission(appContext,
+                Manifest.permission.USE_FINGERPRINT) !=
+                PackageManager.PERMISSION_GRANTED) {
```

Comment:
Not needed.

See: https://developer.android.com/training/permissions/requesting.html

> Normal permissions do not directly risk the user's privacy. If your app lists a normal permission in its manifest, the system grants the permission automatically.